extensions/vscode-api-tests/src/singlefolder-tests/workspace.tasks.test.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+
+suite.only('workspace-namespace', () => {
+
+	suite('Tasks', () => {
+
+		test('CustomExecution2 task should start and shutdown successfully', (done) => {
+			interface CustomTestingTaskDefinition extends vscode.TaskDefinition {
+				/**
+				 * One of the task properties. This can be used to customize the task in the tasks.json
+				 */
+				customProp1: string;
+			}
+			const taskType: string = 'customTesting';
+			const taskName = 'First custom task';
+			const reg1 = vscode.window.onDidOpenTerminal(term => {
+				reg1.dispose();
+				const reg2 = term.onDidWriteData(e => {
+					reg2.dispose();
+					assert.equal(e, 'testing\r\n');
+					term.dispose();
+				});
+			});
+			const taskProvider = vscode.tasks.registerTaskProvider(taskType, {
+				provideTasks: () => {
+					let result: vscode.Task[] = [];
+					let kind: CustomTestingTaskDefinition = {
+						type: taskType,
+						customProp1: 'testing task one'
```

Comment:
It would be great if the test code could demonstrate the use of the value of 'customProp1'.

When I used this in my project, I ended up creating some wrapper classes around the use of virtual terminal process and custom executions so I could get the custom execution data (the properties) back into the execution of the extension code in a seemingly clean way.

When the code is all in one test like this, the "capturing" works just fine, but when you do things like put your task definition in one file, execution of the code in another, etc. it becomes a bit cumbersome.

I can point you to what I ended up doing if you want.

Reply:
Would this be as easy as just adding an assertion somewhere?